src/Microsoft.AspNet.OData.Shared/Formatter/ODataModelBinderConverter.cs

```diff
@@ -344,14 +344,14 @@ internal static object ConvertTo(string valueString, Type type)
             int i = 0;
             foreach (object item in sources)
             {
-                object newItem = CovertResourceId(item, resourceSet.Resources[i].Resource, entityTypeReference,
+                object newItem = CovertResourceId(item, ( (ODataResourceWrapper)resourceSet.Resources[i]).ResourceBase, entityTypeReference,
```

Comment:
>(ODataResourceWrapper) [](start = 58, length = 22)

I assume this cast can't fail because resourceSet is an OdataResourceSetWrapper, not an ODataResourceSetWrapperBase so all Resources will be OdataResourceWrappers (not links, etc.)?